app/services/legacy/beta_sync_service.rb

```diff
@@ -0,0 +1,43 @@
+class Legacy::BetaSyncService
+  ENDPOINTS = {
+    modified_datasets: 'api/3/action/package_search?q=metadata_modified:[NOW-1DAY%20TO%20NOW]&rows=5000'.freeze,
+    new_datasets: 'api/3/action/package_search?q=metadata_created:[NOW-1DAY%20TO%20NOW]&rows=5000'.freeze
+  }
+
+  def initialize(args)
+    @orgs_cache = args[:orgs_cache]
+    @theme_cache = args[:theme_cache]
+    @logger = args[:logger]
+    @legacy_server = args[:legacy_server]
+    @count = 0
+  end
+
+  def run
+    @logger.info "Importing legacy datasets...\r"
+    [modified_datasets, new_datasets].each do |legacy_datasets|
+      legacy_datasets.fetch('result', {}).fetch('results', []).each do |legacy_dataset|
+        import legacy_dataset
+      end
+    end
+    @logger.info "Imported #{@count} datasets...\r"
+  end
+
+  private
+
+  def modified_datasets
+    @legacy_server.get ENDPOINTS[:modified_datasets]
+  end
+
+  def new_datasets
+    @legacy_server.get ENDPOINTS[:new_datasets]
+  end
+
+  def import(legacy_dataset)
+    legacy_dataset_id = legacy_dataset['id']
+    @logger.info "Attempting to save legacy dataset to postgres and elasticsearch - legacy_id: #{legacy_dataset_id}"
+    Legacy::DatasetImportService.new(legacy_dataset,@orgs_cache, @theme_cache).run
+    Legacy::DatasetIndexService.new.index(legacy_dataset['id'])
```

Comment:
@laurentqro - Creating instances of DatasetIndexService and DatasetImportService potentially makes this class difficult to test. An option would be to inject but this will break the contact with DatasetImportService#run and require further edits as a result

Reply:
@laurentqro - update - have managed to get tests working with the above implementation

Reply:
Update - have managed to create to test the BetaSyncService with the current implementation but it relies heavily on mocks - may need to rethink this